pkg/controllers/readiness/controller.go

```diff
@@ -0,0 +1,34 @@
+package readiness
+
+import (
+	"flag"
+	"github.com/ViaQ/log-exploration-api/pkg/configuration"
+	"github.com/ViaQ/log-exploration-api/pkg/elastic"
+	"github.com/ViaQ/log-exploration-api/pkg/middleware"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+func NewReadinessController(router *gin.Engine) {
+	router.GET("/ready", ReadinessHandler)
+}
+
+func ReadinessHandler(gctx *gin.Context) {
+	middleware.AddHeader()
+	esAddress := flag.Lookup("es-addr").Value.(flag.Getter).Get().(string)
```

Comment:
Same here. Please use dependency for the controller (`logsProvider`) instead of creating one inside.